### PR TITLE
[ST] PodSetSt Reconcilation test, checks addition

### PR DESCRIPTION
**Pull request by henryZrncik**
### Type of change

- Refactoring

### Description

checks added to make sure status of `StrimziPodSet` is updated as expected, and clients can produce and consume data after changes take place.

### CODE REVIEW

File: systemtest/src/test/java/io/strimzi/systemtest/operators/PodSetST.java [hunk 1]
```diff
@@ -46,6 +50,16 @@ void testPodSetOnlyReconciliation(ExtensionContext extensionContext) {
         final int replicas = 3;
         final int probeTimeoutSeconds = 6;
 
+        // setup clients
+        KafkaClients clients = new KafkaClientsBuilder()
```
Comment (see-quick) [inline systemtest/src/test/java/io/strimzi/systemtest/operators/PodSetST.java:54]: I am also thinking about continuous sending messages during update of FG? Do you think it is worth @im-konge @henryZrncik ?

**Review by see-quick — APPROVED** [review]
LGTM, just one comment. Anyway looks good! 👍

File: systemtest/src/test/java/io/strimzi/systemtest/operators/PodSetST.java (hunk 1 above)
Comment (henryZrncik) [reply systemtest/src/test/java/io/strimzi/systemtest/operators/PodSetST.java:54]: The changes would be, as some of brokers will go down (not only mentioning reconciliation),   config producer to `ack=al`l, broker to create topics by default with appropriate `min.insync.replicas` and `replicaiton` factor). Similar workaround  was done in `SecurityST testCaRenewalBreakInMiddle`

But I do think it is worth adding there.

Comment (henryZrncik) [reply systemtest/src/test/java/io/strimzi/systemtest/operators/PodSetST.java:56]: I added suggestions into the code base. 👍

**Comment by henryZrncik** [comment]
@strimzi-ci run tests --profile=all --install-type=bundle --cluster-type=ocp --cluster-version=4.14 --testcase=PodSetST

**Review by im-konge — APPROVED** [review]

**Pull request merged**